pycytominer/operations/transform.py

```diff
@@ -67,58 +73,62 @@ def fit(self, X, y=None):
-            U = pd.DataFrame(U * np.sign(np.diag(U)))
+            self.standard_scaler = StandardScaler().fit(X)
+            variances = self.standard_scaler.var_
+            if np.any(variances == 0):
+                raise ValueError(
+                    "Divide by zero error, make sure low variance columns are removed"
+                )
 
-            # Process the eigenvalues into a diagonal matrix and fix rounding errors
-            D = np.diag(1.0 / np.sqrt(s.clip(self.epsilon)))
+            X = self.standard_scaler.transform(X)
+        else:
+            if self.center:
+                self.mean_centerer = StandardScaler(with_mean=True, with_std=False).fit(
+                    X
+                )
+                X = self.mean_centerer.transform(X)
 
-            # Calculate the sphering matrix
-            self.W = np.dot(D, U.transpose())
+        # Get the number of observations and variables
+        n, d = X.shape
 
-            # If ZCA, perform additional rotation
-            if self.method == "ZCA":
-                self.W = np.dot(U, self.W)
+        # compute the rank of the matrix X
+        r = np.linalg.matrix_rank(X)
 
-        if self.method in ["PCA-cor", "ZCA-cor"]:
-            # Get the correlation matrix
-            R = np.corrcoef(X.transpose())
+        assert (r == n - 1) | (r == d), "X is not full rank"
 
-            # Get the eigenvalues and eigenvectors of the correlation matrix
-            try:
-                t, G = eigh(R)
-            except ValueError:
-                raise ValueError(
-                    "Divide by zero error, make sure low variance columns are removed"
-                )
+        # Get the eigenvalues and eigenvectors of the covariance matrix using SVD
+        _, Sigma, Vt = np.linalg.svd(X, full_matrices=True)
 
-            # Fix sign ambiguity of eigenvectors
-            G = pd.DataFrame(G * np.sign(np.diag(G)))
+        # if n <= d then Sigma has shape (n,) so it will need to be expanded to
+        # d filled with the value r'th element of Sigma
+        if n <= d:
+            assert Sigma.shape[0] == n, "Unexpected shape of Sigma"
+            assert r == n - 1, "Unexpected rank"
+            Sigma = np.concatenate((Sigma[0:r], np.repeat(Sigma[r - 1], d - r)))
 
-            # Process the eigenvalues into a diagonal matrix and fix rounding errors
-            D = np.diag(1.0 / np.sqrt(t.clip(self.epsilon)))
+        Sigma = Sigma + self.epsilon
 
-            # process the covariance diagonal matrix and fix rounding errors
-            v = np.diag(1.0 / np.sqrt(np.diag(C).clip(self.epsilon)))
+        self.W = (Vt / Sigma[:, np.newaxis]).transpose() * np.sqrt(n - 1)
 
-            # Calculate the sphering matrix
-            self.W = np.dot(np.dot(D, G.transpose()), v)
+        # If ZCA, perform additional rotation
+        if self.method in ["ZCA", "ZCA-cor"]:
+            # If rank is not d then ZCA is not possible
+            # TODO: Explain this better
```

Comment:
I fear that this would take a while to revisit after we merge, would it take too long to explain?

Reply:
Turns out this was very likely an incorrect requirement 
I've added this comment in the code for our future selves
```
            # Note: There was previously a requirement r==d otherwise the
            # ZCA transform would not be well defined. However, it later appeared
            # that this requirement was not necessary.
```